src/pages/SetupPage.js

```diff
@@ -127,13 +135,19 @@ const SetupPage = () => {
 
   useEffect(() => {
     const { modes, misc } = config
-    const { showMines, showNearbyMines } = misc
+    const { showMines, showNearbyMines, showNumbers } = misc
     setModes(modes)
     setShowMines(showMines)
     setShowNearbyMines(showNearbyMines)
-  }, [])
+    setShowNumbers(showNumbers)
+  }, [config])
 
   useEffect(() => {
+    const getValue = (type) => {
+      if (!modes[selectedMode]) return 0
+      return modes[selectedMode][type]
+    }
+
```

Comment:
Did you find a good reason to move `getValue` into `useEffect`?
My thinking is: `useEffect` will be executed multiple times, so the function will be created multiple times and we can avoid that by moving it out of `useEffect`.
Well... Practically it will be the same.
Maybe I'm just being picky 😅

Reply:
I don't remember exactly why, but `getValue` is defined inside `SetupPage` so it will be instantiated on every render, whereas if it is inside the hook, it will only be instantiated when modes or selectedMode are mutated.

other solutions:

### keep getValue outside the useEffect
```
useEffect(() => {
    setRows(getValue('rows'))
    setCols(getValue('cols'))
    setMines(getValue('mines'))
  }, [modes, selectedMode])
```
this bothers the linter and it is not "stand of the art" react


### add getValue to the dependencies
```
useEffect(() => {
    setRows(getValue('rows'))
    setCols(getValue('cols'))
    setMines(getValue('mines'))
  }, [getValue, modes, selectedMode])
````
getValue would be something new on each render, then this hook will be triggered on each render (bad perf)

### optimal solution: have getValue as an external function
define getValue outside the component with
```
const getValue = (modes, selectedMode, type) => {
  if (!modes[selectedMode]) return 0
  return modes[selectedMode][type]
}
```
and have the useEffect be
```
useEffect(() => {
    setRows(getValue(modes, selectedMode, 'rows'))
    setCols(getValue(modes, selectedMode, 'cols'))
    setMines(getValue(modes, selectedMode, 'mines'))
  }, [modes, selectedMode])
```

but then, at the end, what does getValue do? something fairly simple, almost nothing and its memory impact is very small, so does this worth the complexity? as long as `getValue` stays small, I think the most readable solution is to put it just next to where it is used :-)

This is opinionated, it's your project, you decide!

Have a nice day, old friend, I miss coding with you.

Reply:
Oh, you're right!
To avoid what I was referring to we'll need to move that method outside of the component.
And the current implementation is actually better than the previous one 😁 
Thanks a lot @gasp!
I'm sure we will find more project to work on together 👍